### PR TITLE
Change last_attempt_for_any to use UserLevel rather than Activity.

### PR DESCRIPTION
**Pull request by ashercodeorg**
Note that all usages of `last_attempt_for_any`, specifically in `dashboard/app/controllers/api_controller.rb` and `dashboard/app/controllers/script_levels_controller.rb`, do not need to be updated with this change as the result of them only accessing `level` and `level_source` from the return of `last_attempt_for_any`.

This is one of many (small) steps towards deprecating the activities table. Another PR will update `last_attempt`, though doing so requires updating some of the calling method.

### CODE REVIEW

File: dashboard/test/controllers/api_controller_test.rb [hunk 1]
```diff
@@ -138,27 +146,32 @@ def make_progress_in_section(script)
...
         'stage' => "Stage 1: translation missing: en-us.data.script.name.#{script.name}.#{script.stages[0].name}",
         'puzzle' => 1,
         'question' => 'Text Match 1',
-        'response' => 'answer for student 2',
-        'url' => "http://test.host/s/#{script.name}/stage/1/puzzle/1?section_id=#{@section.id}&user_id=#{@student_2.id}"},
-      ]
+        'response' => 'Here is the answer 2',
+        'url' => "http://test.host/s/#{script.name}/stage/1/puzzle/1?section_id=#{@section.id}&user_id=#{@student_2.id}"
+      }
+    ]
```
Comment (ashercodeorg) [inline dashboard/test/controllers/api_controller_test.rb:174]: These changes to `expected_response` are whitespace changes, except for changes in `response`.

**Comment by joshlory** [comment]
What updates need to happen to the callers of `last_attempt` in order to load that data from `UserLevel`?

**Comment by Hamms** [comment]
LGTM!

**Comment by ashercodeorg** [comment]
The views that rely on `last_attempt` being a `UserLevel` (EDIT: should be `Activity`) need to be updated. Not sure how much work that is, I haven't even looked at how they use it yet.

**Comment by joshlory** [comment]
s/`UserLevel`/`Activity`?

**Comment by ashercodeorg** [comment]
Yes, that should have been `Activity`. Fixed (via EDIT).

**Comment by ashercodeorg** [comment]
Aside: Having investigated a bit more, it looks as if there aren't any strong dependencies on `last_attempt` being an `Activity`. For WIP, see #10162, expecting it to be force pushed after further commits.

**Comment by joshlory** [comment]
LGTM

**Pull request merged**